lib/sdr_client/deposit/default_file_set_builder.rb

```diff
@@ -6,10 +6,12 @@ module Deposit
     class DefaultFileSetBuilder
       # @return [Request] request The initial request
       # @param [Array<SdrClient::Deposit::Files::DirectUploadResponse>] uploads the uploaded files to attach.
+      # @param [Hash<String,Hash<String, String>>] files_metadata filename, hash of additional file metadata.
       # @return [Request] a clone of this request with the uploads added
-      def self.run(request:, uploads: [])
+      def self.run(request:, uploads: [], uploads_metadata: {})
```

Comment:
We aren’t going to use the DefaultFileSetBuilder for GB because it does one fileset per resource and we need one that puts all files with the same name excluding suffix in one fileset. That doesn’t mean this isn’t good stuff though. I’d like your thoughts on how you want to proceed given that information.

Reply:
I'd say let's write that code first and then see if/what refactoring makes sense.

Reply:
It is in https://github.com/sul-dlss/google-books/pull/173